pkg/operation/botanist/secrets.go

```diff
@@ -306,6 +307,32 @@ func (b *Botanist) syncShootCredentialToGarden(
 	return err
 }
 
+func (b *Botanist) deleteSSHKeypair(ctx context.Context) error {
+	if err := b.deleteShootCredentialFromGarden(ctx, gutil.ShootProjectSecretSuffixSSHKeypair); err != nil {
+		return err
+	}
+
+	if err := b.deleteShootCredentialFromGarden(ctx, gutil.ShootProjectSecretSuffixOldSSHKeypair); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (b *Botanist) deleteShootCredentialFromGarden(ctx context.Context, nameSuffix string) error {
```

Comment:
Can you enhance this functions like this?

```suggestion
func (b *Botanist) deleteShootCredentialFromGarden(ctx context.Context, nameSuffixes ...string) error {
```

Then you can simplify the `deleteSSHKeypair` function as well.